### ISSUE TITLE
false positive "error: Type application has too many types (1 expected)" when using pipe instead of Union for type alias.

### ISSUE THREAD

**Issue by tayler6000** [python/mypy#12604]
**Bug Report**

When using a pipe to create a type alias `error: Type application has too many types (1 expected)` is raised.  This is not the case if you use a Union.

**To Reproduce/Actual Behavior**

```python
from typing import Union


type_alias1= tuple[int, int, int] |  int  # error: Type application has too many types (1 expected)
type_alias2 = Union[tuple[int, int, int], int]  # passes type check
```

**My Environment**

- Mypy versions used: 0.960+dev.10ba5c1aa14e01be1cfacb322136e22751617d26, and 0.942
- Python version used: Python 3.10.2
- Operating system and version: Windows 10 Home (21H2)

**Comment by JelleZijlstra** [comment]
Duplicate of #11098
